### DISCUSSION TITLE
Can we use and render custom svelte components?

### DISCUSSION THREAD

**Discussion by nilskj** [jacwright/typewriter#87] [Q&A]
Hello! Is it possible to make a custom line that renders a svelte component instead of html markup? My idea is that of a page builder so I need to insert custom components, example like this:
```js
  import Hero from "./Hero.svelte";

  const editor = (window.editor = new Editor());

  editor.typeset.lines.add({
    name: "hero",
    selector: "hero",
    commands: (editor) => () => editor.toggleLineFormat("hero"),
    render: (attributes, children) => {
      return h(Hero, { ...attributes }, children);
    },
  });
```
I guess the VDOM has no clue how to render that. Is this possible somehow?

**Comment by taylorhadden** [comment]
I do something similar, but I get it done with custom elements that stand up svelte components inside of shadow-dom.

I would heartily support integrating native Svelte constructor support into the VDom implementation.
